ui/apps/platform/src/Containers/Collections/CollectionsTablePage.tsx

```diff
@@ -7,17 +7,66 @@ import {
     Flex,
     FlexItem,
     ButtonVariant,
+    Divider,
+    Alert,
+    Bullseye,
+    Spinner,
 } from '@patternfly/react-core';
 
 import PageTitle from 'Components/PageTitle';
 import LinkShim from 'Components/PatternFly/LinkShim';
 import { collectionsPath } from 'routePaths';
+import useRestQuery from 'Containers/Dashboard/hooks/useRestQuery';
+import { getCollectionCount, listCollections } from 'services/CollectionsService';
+import CollectionsTable from './CollectionsTable';
 
 type CollectionsTablePageProps = {
     hasWriteAccessForCollections: boolean;
 };
 
 function CollectionsTablePage({ hasWriteAccessForCollections }: CollectionsTablePageProps) {
+    const listQuery = useCallback(
+        () => listCollections({}, { field: 'name', reversed: false }, 0, 20),
+        []
+    );
+    const { data: listData, loading: listLoading, error: listError } = useRestQuery(listQuery);
+
+    const countQuery = useCallback(() => getCollectionCount({}), []);
+    const { data: countData, loading: countLoading, error: countError } = useRestQuery(countQuery);
+
+    const isDataAvailable = typeof listData !== 'undefined' && typeof countData !== 'undefined';
+    const isLoading = !isDataAvailable && (listLoading || countLoading);
+    const loadError = listError || countError;
```

Comment:
A question to learn from your experience instead of a change request.

What do you think are pro and con to distinguish errors from primary and secondary requests?
Because I have not yet written a table which has PatternFly pagination, I am not sure how robust it would be to render if list requests succeeds and count request fails.

Reply:
That is a good question. The count data eventually needs to get passed into the PatternFly `<Pagination>` component, but we could probably fallback and do something more graceful than placing the entire section into an "error" state.

It looks like the Pagination `itemCount` prop is optional, which gives us the "of 0" UI:
<img width="863" alt="image" src="https://user-images.githubusercontent.com/1292638/192372640-b9797303-edb6-4780-a4b5-cc53463cb34c.png">
<img width="863" alt="image" src="https://user-images.githubusercontent.com/1292638/192372684-63f96d13-d131-4464-9339-91e111626558.png">

We could also pass `Infinity` to get a similar behavior:
<img width="863" alt="image" src="https://user-images.githubusercontent.com/1292638/192372782-4b5f15cb-9108-4f86-8703-4e9809606a49.png">
<img width="863" alt="image" src="https://user-images.githubusercontent.com/1292638/192372796-7d061056-2dc3-4a69-8d6c-d2c757657915.png">

Note that in both these cases the pagination still works correctly, with the exception of allowing the user to "overpage".

PF also provides a `toggleTemplate` option we could use, if we really wanted to be robust, which lets us specify what is shown in the case of a `NaN` `itemCount`:
```typescript
toggleTemplate={({ firstIndex, lastIndex, itemCount, ofWord }) => {
  // When itemCount is not an integer
  // TODO Implement happy path
  return (
    <span>
      <b>{firstIndex}</b>-<b>{lastIndex}</b> {ofWord}{' '}
      <b>unknown</b>
    </span>
  );
}}
```
<img width="863" alt="image" src="https://user-images.githubusercontent.com/1292638/192373466-6b3b70a9-743d-45e3-8259-c023b2351b38.png">

If this is something we wanted to do, it might actually be one of the cases where we would want a wrapper around a PF component so that we could get this on other pages that also use PF pagination, since it would be a bit verbose to write it out each time.

Reply:
Yeah, allowing overpage seems like a more serious problem, therefore leave well enough alone for now.

Although there is a prop to customize the count with text, it seems much cost for small benefit:
https://www.patternfly.org/v4/components/pagination#indeterminate